### PR TITLE
feat: add technical architecture info

### PR DESCRIPTION
**Pull request by kpopper**
Adds a new section to our public Github readme to share a basic overview of our technical architecture to interested outsiders (for example people considering coming to work at Oak).

### CODE REVIEW

File: profile/readme.md [hunk 1]
```diff
@@ -17,3 +17,11 @@ We also openly licence our code where it is possible to do so.
 ### Oak Web Application
 
 Our main public-facing application is the open source [Oak Web Application](https://github.com/oaknational/Oak-Web-Application), built in [Next.js](https://nextjs.org/), [React](https://react.dev/), [Styled-Components](https://styled-components.com/) and using the [Oak React components library](https://github.com/oaknational/oak-components).
+
+## Our approach to our technical architecture
+
+We take a serverless-first approach to hosting, with the majority of our applications and services running on dedicated PaaS and IaaS provider platforms.
```
Comment (kpopper) [inline profile/readme.md]:
```suggestion
We take a serverless-first approach to hosting, with the majority of our applications and services running on dedicated Platform-as-a-Service (PaaS) and Infrastructure-as-a-Service (IaaS) provider platforms. 
```

**Review by johnrobeds — APPROVED** [review]

File: profile/readme.md [hunk 2]
```diff
@@ -17,3 +17,11 @@ We also openly licence our code where it is possible to do so.
...
 
 Our main public-facing application is the open source [Oak Web Application](https://github.com/oaknational/Oak-Web-Application), built in [Next.js](https://nextjs.org/), [React](https://react.dev/), [Styled-Components](https://styled-components.com/) and using the [Oak React components library](https://github.com/oaknational/oak-components).
+
+## Our approach to our technical architecture
+
+We take a serverless-first approach to hosting, with the majority of our applications and services running on dedicated Platform-as-a-Service (PaaS) and Infrastructure-as-a-Service (IaaS) provider platforms. 
+
+Our web applications are built and hosted on [Netlify](https://www.netlify.com/), with a handful on [Vercel](https://vercel.com/), while many of our supporting services are either outsourced or run on [Google Cloud Functions](https://cloud.google.com/functions/?hl=en). 
+
+We are in the process of moving all of our infrastructure management to [Terraform](https://github.com/oaknational/Cloud-Config).
```
Comment (triwats) [inline profile/readme.md]: Right now, Cloud Config (https://github.com/oaknational/Cloud-Config) is private. Not sure if we wanna omit the link? 

Part of the work I've been doing works to make this project potentially public, I could add this to a new Task in Notion to do that as part of that Epic?

**Review by triwats — CHANGES_REQUESTED** [review]

File: profile/readme.md (hunk 2 above)
Comment (johnrobeds) [reply profile/readme.md]: I'd omit the link whilst it's private. I'm happy for this to be public, providing we're comfortable with all the usual security caveats (i.e no risk of legacy secrets in there etc)

Comment (kpopper) [reply profile/readme.md]:
```suggestion
We are in the process of moving all of our infrastructure management to [Terraform](https://www.terraform.io/). 
```

**Review by triwats — APPROVED** [review]

**Comment by kpopper** [comment]
I'm overwriting the "signed commits" protection. I know the commit is mine and have set it up for future commits. I think the history in the PR above is more important than the signed commit here.

**Pull request merged**